pycqed/analysis_v2/base_analysis.py

```diff
@@ -634,8 +635,16 @@ def save_figures(self, savedir: str = None, savebase: str = None,
                 savename = os.path.join(savedir, savebase + key + tstag + '.' + fmt)
                 self.figs[key].savefig(savename, bbox_inches='tight',
                                        fmt=fmt, dpi=dpi)
-            if close_figs:
-                plt.close(self.figs[key])
+        if close_figs:
+            self.close_figs(key_list)
+
+    def close_figs(self, key_list='auto'):
+        if key_list == 'auto' or key_list is None:
+            key_list = self.figs.keys()
+        for key in list(key_list):
+            plt.close(self.figs[key])
+            self.figs.pop(key)
+            self.axs.pop(self.plot_dicts[key]['fig_id'])
```

Comment:
@antsr This change makes CryoscopeAnalysis crash

Reply:
I fixed and tested it.